### ISSUE TITLE
Must check for state UndefType before trying to convert to other type

### ISSUE THREAD

**Issue by seime** [seaside1/jrule#71]
java.lang.NumberFormatException: Invalid BigDecimal value: UNDEF
        at org.openhab.core.library.types.QuantityType.<init>(QuantityType.java:130) ~[?:?]
        at org.openhab.core.library.types.QuantityType.<init>(QuantityType.java:106) ~[?:?]
        at org.openhab.core.library.types.QuantityType.valueOf(QuantityType.java:206) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.excutioncontext.JRuleItemExecutionContext.lambda$4(JRuleItemExecutionContext.java:67) ~[?:?]
        at java.util.Optional.filter(Optional.java:223) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.excutioncontext.JRuleItemExecutionContext.matchCondition(JRuleItemExecutionContext.java:67) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.excutioncontext.JRuleItemChangeExecutionContext.match(JRuleItemChangeExecutionContext.java:51) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.JRuleEngine.lambda$46(JRuleEngine.java:267) ~[?:?]

**Comment by querdenker2k** [comment]
should be fixed in #77 . The error triggered me there as well.

**Comment by seaside1** [comment]
After merge of PR 66 I see this with various types:
```
2022-11-05 20:37:19.402 [WARN ] [ab.core.internal.events.EventHandler] - Dispatching/filtering event for subscriber 'org.openhab.core.e
vents.EventSubscriber' failed: Invalid BigDecimal value: ON
java.lang.NumberFormatException: Invalid BigDecimal value: ON
        at org.openhab.core.library.types.QuantityType.<init>(QuantityType.java:130) ~[?:?]
        at org.openhab.core.library.types.QuantityType.<init>(QuantityType.java:106) ~[?:?]
        at org.openhab.core.library.types.QuantityType.valueOf(QuantityType.java:206) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.excutioncontext.JRuleItemExecutionContext.lambda$4(JRuleItemExecutionContext.java:69) ~[?:?]
        at java.util.Optional.filter(Optional.java:223) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.excutioncontext.JRuleItemExecutionContext.matchCondition(JRuleItemExecutionContext.java:69) ~[?:?]
        at org.openhab.automation.jrule.internal.engine.excutioncontext.JRuleItemReceivedUpdateExecutionContext.match(JRuleItemReceived

```

I'll see if I can find more stacktraces

**Comment by seaside1** [comment]
I have a precondition rule with a number item being check to trigger this:

 @JRuleWhenItemReceivedUpdate(item = MyScene.ITEM, condition =  @JRuleCondition(gt = 20))

**Comment by seaside1** [comment]
And looks better when I merge pr77 with main. Still testing it.

**Comment by seime** [comment]
Have not seen this issue on my side after merging and running locally since the weekend

**Comment by seime** [comment]
Solved by #77
